src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { Fragment } from 'react';
+import cx from 'classnames';
+import { useThemeContext } from '~/contexts';
+import type { RatingType } from './Rating.types';
+import styles from './Rating.module.css';
+
+// Array generator returns array of string
+// eg: ['1', '2', '3', '4', '5']
+type ArrayGenerator = (length: number) => string[];
+const arrayGenerator: ArrayGenerator = (maxItems) =>
+  [...Array(maxItems).keys()].map((item) => (item + 1).toString());
+
+const Rating: RatingType = ({
+  className,
+  theme = 'dark',
+  name,
+  value,
+  onChange,
+  dataTestRef,
+  errorMessage,
+  max = 5,
+}) => {
+  const currentTheme = useThemeContext(theme, 'dark');
```

Comment:
The default value is not needed, right?

```suggestion
  const currentTheme = useThemeContext(theme);
```

Reply:
yeah that is correct. it can be dark by default